src/main/java/org/jenkinsci/plugins/github/webhook/BackupForCommitWebhook.java

```diff
@@ -0,0 +1,124 @@
+package org.jenkinsci.plugins.github.webhook;
+
+import com.cloudbees.jenkins.GitHubPushTrigger;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.AbstractProject;
+import hudson.model.AperiodicWork;
+import hudson.model.AsyncAperiodicWork;
+import hudson.model.TaskListener;
+import hudson.slaves.EnvironmentVariablesNodeProperty;
+import hudson.slaves.NodeProperty;
+import hudson.slaves.NodePropertyDescriptor;
+import hudson.triggers.SCMTrigger;
+import hudson.util.DescribableList;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * In case of unreachable CI due to n/w issue or CI corruption, web hook payloads will
+ * not reach CI and build won't trigger for new commit(s). Below class serves as a safe
+ * guard mechanism to make sure CI jobs (GitHub web hook trigger enabled) don’t miss any
+ * commits by polling for new commits on certain conditions.
+ */
+@Extension
+public class BackupForCommitWebhook extends AsyncAperiodicWork {
+
+    private static final Logger LOGGER = getLogger(BackupForCommitWebhook.class);
+
+    private static BackupForCommitWebhook currentTask;
+    private static final int LAST_BUILD_TIME_THRESHOLD = 45 * 60 * 1000;
+    private static final int RECURRENCE_PERIOD = 60 * 60 * 1000;
+
+
+    public BackupForCommitWebhook() {
+        super("Check for new commits");
+    }
+
+    @Override
+    protected void execute(TaskListener taskListener) throws IOException, InterruptedException {
+        LOGGER.info("Executing BackupForCommitWebhook");
+        if (isBackupForCommitWebhookEnabled()) {
+            List<AbstractProject> jobs = Jenkins.getInstance().getItems(AbstractProject.class);
+            for (AbstractProject job : jobs) {
+                if (isEligible(job)) {
+                    if (job.poll(taskListener).hasChanges()) {
```

Comment:
Yes, `poll` is abstractproject->gitscm functionality. Obviously you want to run periodic poll on repository that is done via gitscm (git-plugin). I will repeat in 100th time that github-plugin now do only triggering via github hook -> gitscm.poll(). So the right place for periodic run is git-plugin. But job ALREADY has 'build periodically commits" that is generic cron based configurable trigger for project.poll(). So you don't need anything to do at all beside enable checkbox.